sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -104,77 +210,194 @@ def hpat_pandas_series_getitem(self, idx):
+        (isinstance(idx, (str, types.UnicodeType, types.StringLiteral)) and index_is_string)):
+        def hpat_pandas_series_getitem_index_impl(self, idx):
+            mask = numpy.empty(len(self._data), numpy.bool_)
+            for i in numba.prange(len(self.index)):
+                mask[i] = self.index[i] == idx
+            return pandas.Series(self._data[mask], self.index[mask])
 
-        return hpat_pandas_series_getitem_idx_integer_impl
+        return hpat_pandas_series_getitem_index_impl
+
+    if (isinstance(idx, types.Integer) and index_is_string):
+        def hpat_pandas_series_idx_impl(self, idx):
+            return self._data[idx]
+
+        return hpat_pandas_series_idx_impl
 
     if isinstance(idx, types.SliceType):
+    #Return slice for str values not implement
         def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc2
-            """
-
-            result = pandas.Series(self._data[idx])
-            return result
+            return pandas.Series(self._data[idx], self.index[idx])
 
         return hpat_pandas_series_getitem_idx_slice_impl
 
-    if isinstance(idx, SeriesType):
+    if isinstance(idx, (types.List, types.Array)):
+        def hpat_pandas_series_getitem_idx_list_impl(self, idx):
+            if (idx[0] == True or idx[0] == False):
+                return pandas.Series(self._data[idx], self.index[idx])
+        return hpat_pandas_series_getitem_idx_list_impl
+
+    if (isinstance(self.index, types.NoneType) and isinstance(idx, SeriesType)):
+        def hpat_pandas_series_getitem_idx_list_impl(self, idx):
+            if (idx._data[0] == True or idx._data[0] == False):
+                index = numpy.arange(len(self._data))
+                return pandas.Series(self._data[idx._data], index[idx._data])
+            res = numpy.copy(self._data[:len(idx._data)])
+            index = numpy.arange(len(self._data))
+            for i in numba.prange(len(res)):
+                for j in numba.prange(len(index)):
+                    if index[j] == idx._data[i]:
+                        res[i] = self._data[j]
+            return pandas.Series(res, index[idx._data])
+        return hpat_pandas_series_getitem_idx_list_impl
+
+    if (isinstance(idx, SeriesType) and not isinstance(self.index, types.NoneType)):
+    #Series with str index not implement
         def hpat_pandas_series_getitem_idx_series_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_setitem_series_bool2
-            """
-            super_index = idx._data
-            result = self._data[super_index]
-            return result
+            if (idx._data[0] is True or idx._data[0] is False):
+                return pandas.Series(self._data[idx._data], self._index[idx._data])
+            res = numpy.copy(self._data[:len(idx._data)])
```

Comment:
That's not valid.
```
s = pd.Series([0, 1, 2], [0, 0, 0])
print(s[[0]])
0    0
0    1
0    2
dtype: int64
```

So, output size is larger than size of idx._data

Also, i think it would be better to use dict here